### ISSUE TITLE
Datadog::Tracer.trace always returns the span, even if block is given

### ISSUE THREAD

**Issue by ejholmes** [DataDog/dd-trace-rb#41]
When [Datadog::Tracer.trace](https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/tracer.rb#L116) is called with a block, I would expect the last line of the block to be used as a return value.

For example, given the following:

```ruby
def call(env)
  @tracer.trace('rack.request') do |span|
    @app.call(env)
  end
end
```

I would expect `@trace.trace` to return the rack response. In reality, it currently returns the Datadog::Span.

**Comment by palazzem** [comment]
Thank you for pointing that and for your contribution! I'll check the PR soon!

**Comment by palazzem** [comment]
Fixed in #42
